internal/api/client/client.go

```diff
@@ -249,6 +251,50 @@ func Update(ctx context.Context, c *Client, obj APIObject) error {
 	return nil
 }
 
+// SignAndUpdate signs the given apiObject and updates the object and it's signature in the API.
+// This function should be used in favor of manually managing signatures.
+// The signing is specified as part of the signer argument.
+func SignAndUpdate(ctx context.Context, c *Client, obj APIObject, signer sigstore.Signer) error {
```

Comment:
can you use the new code inside the attestationconfigapi client?

Reply:
Ok, but I think it makes sense to test the design of an API by actually using it. How do we know that the new code even works?

Reply:
We don't. But it's easier to review this way. Since the code isn't used I don't see a problem with that. I will start using the code and thus test it shortly after it's merged.